src/main/ir/ast/PrintType.scala

```diff
@@ -2,16 +2,23 @@ package ir.ast
 
 import ir.Type
 import ir.interpreter.Interpreter._
+import utils.paternoster.visualisation.TypeVisualiser
 
 /**
   * A pattern for debugging Lift code.
   * Identity function that prints the Lift type of its input.
   * Generates no OpenCL code.
   */
-case class PrintType() extends Pattern(arity = 1) {
+case class PrintType(visual: Boolean = false, render: Boolean = false,expression:String = "") extends Pattern(arity = 1) {
```

Comment:
I would suggest creating a small algebraic data type for specifying the difference between text or visual output.
I.e. something along the lines
```
sealed trait TypeOutput
case class VisualOutput(render: Boolean, expr: String) extends TypeOutput
case object TextOutput extends TypeOutput
```

This has the advantage that it encapsulates the parameters needed for the visual output properly.
I mean it seems that the second and third argument is ignored if the first is false, but that really doesn't get clear currently (but would with the algebraic data type).